packages/devtools_app/lib/src/debugger/flutter/debugger_model.dart

```diff
@@ -182,3 +184,64 @@ class StackFrameAndSourcePosition {
 
   int get column => sourcePosition?.column;
 }
+
+class Variable extends TreeNode<Variable> {
+  Variable._(this.boundVar);
+
+  factory Variable.create(BoundVariable variable) {
+    return Variable._(variable);
+  }
+
+  BoundVariable boundVar;
+
+  bool get hasChildren =>
+      boundVar.value is InstanceRef &&
+      (boundVar.value as InstanceRef).valueAsString == null;
+
+  Object get value => boundVar.value;
+
+  String get displayValue {
+    final name = boundVar.name;
+    final value = this.value;
+
+    String valueStr;
+    if (value is InstanceRef) {
+      if (value.valueAsString == null) {
+        valueStr = value.classRef.name;
+      } else {
+        valueStr = value.valueAsString;
+        if (value.valueAsStringIsTruncated) {
+          valueStr += '...';
+        }
+        if (value.kind == InstanceKind.kString) {
+          valueStr = "'$valueStr'";
+        }
+      }
+
+      if (value.kind == InstanceKind.kList) {
+        valueStr = '[${value.length}] $valueStr';
+      } else if (value.kind == InstanceKind.kMap) {
+        valueStr = '{ ${value.length} } $valueStr';
+      } else if (value.kind != null && value.kind.endsWith('List')) {
+        // Uint8List, Uint16List, ...
+        valueStr = '[${value.length}] $valueStr';
+      }
+    } else if (value is Sentinel) {
+      valueStr = value.valueAsString;
+    } else if (value is TypeArgumentsRef) {
+      valueStr = value.name;
+    } else {
+      valueStr = value.toString();
+    }
+
+    return '$name: $valueStr';
```

Comment:
When we do implement the `Variables` widget (below), we can use some style differences to call out the variable name and value differently - we won't have to just concatenate them w/ a `:` separator.